pkg/build/builder/dockerutil.go

```diff
@@ -49,11 +50,20 @@ func retryImageAction(actionName string, action func() error) error {
 		time.Sleep(DefaultPushOrPullRetryDelay)
 	}
 
-	switch errors.Cause(err) {
-	case idocker.ErrUnauthorizedForCredentials:
-		// strip off wrappers that mainly add the image name as their added context
-		err = errors.Cause(err)
+	processError := func(err error) error {
+		switch errors.Cause(err) {
+		case idocker.ErrUnauthorizedForCredentials:
+			// strip off wrappers that mainly add the image name as their added context
+			return errors.Cause(err)
+		}
+		return err
+	}
```

Comment:
Please make this a standalone function - could be useful in other contexts

Reply:
Okay, changing both the check for the specific error and the iterating through MultiErrors.  If you only want the unwrap broken out, I'm happy to save that for `retryImageAction()`.

Reply:
I was thinking of just breaking out the unwrap. The multiError seems specific to the docker client.

Reply:
Okay, I think that's got it.  The multiple-errors case turned out to be a different type than I'd thought.